mobile_packet_verifier/src/burner.rs

```diff
@@ -36,7 +36,58 @@ impl<S> Burner<S>
 where
     S: SolanaNetwork,
 {
-    pub async fn burn(&self, pool: &Pool<Postgres>) -> anyhow::Result<()> {
+    pub async fn confirm_and_burn(&self, pool: &PgPool) -> anyhow::Result<()> {
+        self.confirm_pending_txns(pool)
+            .await
+            .context("confirming pending txns")?;
+        self.burn(pool).await.context("burning")?;
+        Ok(())
+    }
+
+    pub async fn confirm_pending_txns(&self, pool: &PgPool) -> anyhow::Result<()> {
+        let pending_txns = pending_txns::fetch_all_pending_txns(pool).await?;
+        tracing::info!(count = pending_txns.len(), "confirming pending txns");
+
+        for pending in pending_txns {
+            // Sleep for at least a minute since the time of submission to
+            // give the transaction plenty of time to be finalized
+            let time_since_submission = Utc::now() - pending.time_of_submission;
+            if Duration::minutes(1) > time_since_submission {
+                let delay = Duration::minutes(1) - time_since_submission;
+                tracing::info!(?pending, %delay, "waiting to confirm pending txn");
+                tokio::time::sleep(delay.to_std()?).await;
+            }
+
+            let signature = pending.signature;
+            let confirmed = self.solana.confirm_transaction(&signature).await?;
+            tracing::info!(?pending, confirmed, "confirming pending txn");
+
+            if confirmed {
+                let sessions =
+                    pending_txns::get_pending_data_sessions_for_signature(pool, &signature).await?;
+                for session in sessions {
+                    let _write = self
+                        .valid_sessions
+                        .write(ValidDataTransferSession::from(session), &[])
+                        .await?;
+                }
+
+                pending_txns::remove_pending_txn_success(pool, &signature).await?;
+            } else {
+                pending_txns::remove_pending_txn_failure(pool, &signature).await?;
+            }
+        }
+
+        Ok(())
+    }
+
+    pub async fn burn(&self, pool: &PgPool) -> anyhow::Result<()> {
+        let pending_txns = pending_txns::pending_txn_count(pool).await?;
```

Comment:
Is this still needed here, since you are confirming right before burn is called?

Reply:
It's technically not needed. I still kind of like it there for testing purposes and making the expectation clear.
I can remove it if you think it should be though.

Reply:
Exposing only the `confirm_and_burn()` functionality is possible, but the changes in tests that would be required to make it clear what functionality is being tested, even when we're willing to use only side effects for confirmation, feel... not juice that isn't worth the squeeze.

In consideration that we're planning on refactoring these Burners in the near future, we've agreed this check is fine where it is, and time is better spent elsewhere at this time.